### PR TITLE
SecurityAdvisor1.0

### PR DESCRIPTION
**Pull request by r-puri**
## Status
Ready

## Related Issues
None

## Description
Submitting a new integration for SecurityAdvisor for Demisto

## Screenshots
Paste here any images that will help the reviewer

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------

## Required version of Demisto
4.5.0

## Does it break backward compatibility?
   - No. First release

## Must have
- [ X] Tests
- [ X] Documentation (https://github.com/SecurityAdvisorBot/content/blob/pr/securityadvisor/Integrations/SecurityAdvisor/SecurityAdvisor_description.md)
- [ X] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] Dependency 1
- [ ] Dependency 2
- [ ] Dependency 3

## Additional changes
Describe additional changes done, for example adding a function to common server.

## Technical writer review
Mention and link to the files that require a technical writer review.
- [x ] [YAML file](https://github.com/SecurityAdvisorBot/content/blob/pr/securityadvisor/Integrations/SecurityAdvisor/SecurityAdvisor.yml)
- [x ] [CHANGELOG](https://github.com/SecurityAdvisorBot/content/blob/pr/securityadvisor/Integrations/SecurityAdvisor/CHANGELOG.md)

### CODE REVIEW

**Comment by r-puri** [comment]
Please let me know why the pull-check is failing.

**Comment by Itay4** [comment]
@r-puri `pull-check` will be resolved once our technical writers will finish reviewing the PR

File: Integrations/SecurityAdvisor/Pipfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+certifi = "==2017.11.5"
+chardet = "==3.0.4"
+idna = "==2.6"
+olefile = "==0.44"
+requests = "==2.18.4"
+urllib3 = "==1.22"
+PyYAML = "==3.12"
+
+[requires]
+python_version = "2.7"
```
Comment (anara123) [inline Integrations/SecurityAdvisor/Pipfile]: use python_version 3.7, python2 will be not supported soon

File: Integrations/SecurityAdvisor/SecurityAdvisor_description.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+## Overview
```
Comment (anara123) [inline Integrations/SecurityAdvisor/SecurityAdvisor_description.md]: This file should be name `README.md` 
`SecurityAdvisor_description.md` is detailed description about the integration and should only contain info on how to configure the integration instance and where do I get the API KEY.

```
To get your  __API KEY__ log into www.securityadvisor.io and get your key from "My Profile" section or contact us at support@securityadvisor.io
```

File: Integrations/SecurityAdvisor/SecurityAdvisor_description.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+##### Command Example
+`coach-end-user user="track@securityadvisor.io" context="phishing"`
+
+##### Context Example
+```
+{
+    "SecurityAdvisor.CoachUser": {
+        "coaching_date": "2019-10-04T21:04:19.480425", 
+        "coaching_status": "Pending",
```
Comment (anara123) [inline Integrations/SecurityAdvisor/SecurityAdvisor_description.md]: so if there is a status, how the user should know if coaching status is completed?

File: Integrations/SecurityAdvisor/SecurityAdvisor_test.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import SecurityAdvisor
+import unittest
+from mock import patch, Mock
+
+URL_SUFFIX = 'apis/coachuser/'
```
Comment (anara123) [inline Integrations/SecurityAdvisor/SecurityAdvisor_test.py]: we use pytest to run our tests.
So please rewrite the unit tests to fit pytest format

File: Integrations/SecurityAdvisor/SecurityAdvisor.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
+import demistomock as demisto
+from CommonServerPython import *
+import requests
+import json
+import collections
+# disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+BASE_URL = demisto.params().get('url', 'https://www.securityadvisor.io')
```
Comment (anara123) [inline Integrations/SecurityAdvisor/SecurityAdvisor.py]: please reformat the integration according to our new code conventions
https://github.com/demisto/content/blob/master/docs/code_conventions/README.MD

File: Integrations/SecurityAdvisor/SecurityAdvisor.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        coach_end_user_command()
+    elif demisto.command() == 'test-module':
+        user = 'track@securityadvisor.io'
+        context = "malware"
+        send_message(user, context)
+        demisto.results('ok')
+except Exception, e:
+    demisto.debug('Error in SecurityAdvisor')
+    LOG(e.message)
+    LOG.print_log()
```
Comment (anara123) [inline Integrations/SecurityAdvisor/SecurityAdvisor.py]: remove 
```
LOG(e.message)
LOG.print_log()
```
return_error() will call them both

**Comment by r-puri** [comment]
Hi, can you please help me resolve the failing CI build. Its complaining of not finding a backwards version of SecurityAdvisor.yml. Since this is our first submission there isn't one in the demisto/content master branch. What am I doing wrong?

File: Integrations/SecurityAdvisor/Pipfile (hunk 1 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/Pipfile]: This is updated

File: Integrations/SecurityAdvisor/SecurityAdvisor_description.md (hunk 2 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor_description.md]: This is updated

File: Integrations/SecurityAdvisor/SecurityAdvisor_description.md (hunk 3 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor_description.md]: Currently the user will logon to securityadvisor.io to see detailed reporting. We will be working on importing data to demisto and building dashboards shortly.

File: Integrations/SecurityAdvisor/SecurityAdvisor_test.py (hunk 4 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor_test.py]: This is done.

File: Integrations/SecurityAdvisor/SecurityAdvisor.py (hunk 6 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor.py]: This is done.

File: Integrations/SecurityAdvisor/SecurityAdvisor.py (hunk 5 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor.py]: This is done.

File: Integrations/SecurityAdvisor/SecurityAdvisor_description.md (hunk 2 above)
Comment (anara123) [reply Integrations/SecurityAdvisor/SecurityAdvisor_description.md]: Sorry ,probably I didnt explained myself well.
YOu should have 2 files.
1. `README.md` which contains the full integration documentation - follow https://github.com/demisto/content/blob/master/docs/integration_documentation/README.MD
2. SecurityAdvisor_description.md - should contain only info on how to configure the integration instance and where do I get the API KEY.

File: Integrations/SecurityAdvisor/SecurityAdvisor.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        if demisto.command() == 'coach-end-user':
+            return_outputs(*coach_end_user_command(client, demisto.args()))
+        elif demisto.command() == 'test-module':
+            test_module(client)
+            demisto.results('ok')
+    # Log exceptions
+    except Exception as e:
+        return_error('Failed to execute %s command. Error: %s' %
+                     (demisto.command(), str(e)))
+    LOG.print_log()
```
Comment (anara123) [inline Integrations/SecurityAdvisor/SecurityAdvisor.py]: remove LOG.print_log()

File: Integrations/SecurityAdvisor/SecurityAdvisor_description.md (hunk 2 above)
Comment (anara123) [reply Integrations/SecurityAdvisor/SecurityAdvisor_description.md]: So this file should be named README.md and the other one SecurityAdvisor_description.md

File: Integrations/SecurityAdvisor/SecurityAdvisor.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        readable_output,
+        outputs,
+        result  # raw response - the original response
+    )
+
+
+def test_module(client):
+    """Test Module when testing integration"""
+    data = json.dumps(
+        {"username": "track@securityadvisor.io", "context": "malware"})
```
Comment (anara123) [inline Integrations/SecurityAdvisor/SecurityAdvisor.py]: does "track@securityadvisor.io" user always in the system? I mean, can we rely on the fact that the test will work for any you customer?

Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor.py]: yes, this default user is always present.

File: Integrations/SecurityAdvisor/SecurityAdvisor.py (hunk 7 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor.py]: done

File: Integrations/SecurityAdvisor/SecurityAdvisor_description.md (hunk 2 above)
Comment (r-puri) [reply Integrations/SecurityAdvisor/SecurityAdvisor_description.md]: done

**Comment by r-puri** [comment]
Hi, I have the requested changes. However i dont see the circle ci build running for the last check-in and also cannot access circle-ci dashboard from github. Maybe the service is down. Will try to run in a few minutes again.

**Comment by anara123** [comment]
@kirbles19 please review

https://github.com/SecurityAdvisorBot/content/blob/pr/securityadvisor/Integrations/SecurityAdvisor/README.md
https://github.com/SecurityAdvisorBot/content/blob/pr/securityadvisor/Integrations/SecurityAdvisor/SecurityAdvisor.yml
https://github.com/SecurityAdvisorBot/content/blob/pr/securityadvisor/Integrations/SecurityAdvisor/SecurityAdvisor_description.md

**Comment by r-puri** [comment]
> Typo on main Overview "For example, a user who's system" > "whose"

Updated.

**Comment by r-puri** [comment]
Hi Team, any update on this review?

**Comment by anara123** [comment]
@r-puri sorry for late reply. My bad, should have assign @kirbles19 , so he didn't get the notification

**Comment by kirbles19** [comment]
@Itay4 @anara123 Done.

**Pull request merged**